NullPointerException when updating contents of sitemap

I see this NPE when I edit the contents of my sitemap file.
```
2017-07-09 08:35:52.874 [ERROR] [.core.internal.folder.FolderObserver] - Error handling update of file '/opt/openhab2/conf/sitemaps/1410r.sitemap': null.
java.lang.NullPointerException
        at org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener.getAllItems(PageChangeListener.java:125)[121:org.eclipse.smarthome.io.rest.sitemap:0.9.0.201706270841]
        at org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener.updateItemsAndWidgets(PageChangeListener.java:75)[121:org.eclipse.smarthome.io.rest.sitemap:0.9.0.201706270841]
        at org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener.sitemapContentChanged(PageChangeListener.java:217)[121:org.eclipse.smarthome.io.rest.sitemap:0.9.0.201706270841]
        at org.eclipse.smarthome.io.rest.sitemap.SitemapSubscriptionService.modelChanged(SitemapSubscriptionService.java:277)[121:org.eclipse.smarthome.io.rest.sitemap:0.9.0.201706270841]
        at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.notifyListeners(ModelRepositoryImpl.java:286)[126:org.eclipse.smarthome.model.core:0.9.0.201706270841]
        at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.addOrRefreshModel(ModelRepositoryImpl.java:136)[126:org.eclipse.smarthome.model.core:0.9.0.201706270841]
        at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.checkFile(FolderObserver.java:234)[126:org.eclipse.smarthome.model.core:0.9.0.201706270841]
        at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.processWatchEvent(FolderObserver.java:297)[126:org.eclipse.smarthome.model.core:0.9.0.201706270841]
        at org.eclipse.smarthome.core.service.WatchQueueReader.run(WatchQueueReader.java:206)[102:org.eclipse.smarthome.core:0.9.0.201706270841]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
```
Here's what I've been able to determine so far:

- If I edit the sitemap file (using vi) when ssh'ed into my openhab box, the exception DOES NOT occur.  In the log I see this:
```
2017-07-09 08:49:04.147 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model '1410r.sitemap'
2017-07-09 08:49:04.207 [INFO ] [el.core.internal.ModelRepositoryImpl] - Refreshing model '1410r.sitemap'
2017-07-09 08:49:04.265 [INFO ] [el.core.internal.ModelRepositoryImpl] - Refreshing model '1410r.sitemap'
```

- If I edit the sitemap file (using notepad++ and VS Code) from another box where the file is exposed through a Samba share, the exception DOES occur.  In the log I see this:
```
2017-07-09 08:50:23.768 [INFO ] [el.core.internal.ModelRepositoryImpl] - Refreshing model '1410r.sitemap'
2017-07-09 08:50:23.769 [WARN ] [el.core.internal.ModelRepositoryImpl] - Configuration model '1410r.sitemap' is either empty or cannot be parsed correctly!
2017-07-09 08:50:23.770 [ERROR] [.core.internal.folder.FolderObserver] - Error handling update of file '/opt/openhab2/conf/sitemaps/1410r.sitemap': null.
java.lang.NullPointerException
        at org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener.getAllItems(PageChangeListener.java:125)[121:org.eclipse.smarthome.io.rest.sitemap:0.9.0.201706270841]
        at org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener.updateItemsAndWidgets(PageChangeListener.java:75)[121:org.eclipse.smarthome.io.rest.sitemap:0.9.0.201706270841]
        at org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener.sitemapContentChanged(PageChangeListener.java:217)[121:org.eclipse.smarthome.io.rest.sitemap:0.9.0.201706270841]
        at org.eclipse.smarthome.io.rest.sitemap.SitemapSubscriptionService.modelChanged(SitemapSubscriptionService.java:277)[121:org.eclipse.smarthome.io.rest.sitemap:0.9.0.201706270841]
        at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.notifyListeners(ModelRepositoryImpl.java:286)[126:org.eclipse.smarthome.model.core:0.9.0.201706270841]
        at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.addOrRefreshModel(ModelRepositoryImpl.java:136)[126:org.eclipse.smarthome.model.core:0.9.0.201706270841]
        at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.checkFile(FolderObserver.java:234)[126:org.eclipse.smarthome.model.core:0.9.0.201706270841]
        at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.processWatchEvent(FolderObserver.java:297)[126:org.eclipse.smarthome.model.core:0.9.0.201706270841]
        at org.eclipse.smarthome.core.service.WatchQueueReader.run(WatchQueueReader.java:206)[102:org.eclipse.smarthome.core:0.9.0.201706270841]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
2017-07-09 08:50:23.909 [WARN ] [el.core.internal.ModelRepositoryImpl] - Configuration model '1410r.sitemap' has errors, therefore ignoring it: [627,134]: mismatched input '<EOF>' expecting '}'

2017-07-09 08:50:23.931 [WARN ] [el.core.internal.ModelRepositoryImpl] - Configuration model '1410r.sitemap' has errors, therefore ignoring it: [627,134]: mismatched input '<EOF>' expecting '}'

2017-07-09 08:50:23.971 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model '1410r.sitemap'
2017-07-09 08:50:24.112 [INFO ] [el.core.internal.ModelRepositoryImpl] - Refreshing model '1410r.sitemap'
```

Using inotifywait on the conf/sitemaps directory, I can see that the file is updated differently depending on whether it's edited locally versus through samba.  I suppose this difference is what's causing the issue.

--- a/bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java
+++ b/bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java
@@ -203,12 +203,8 @@ public class SitemapSubscriptionService implements ModelRepositoryChangeListener
         PageChangeListener listener = pageChangeListeners.get(getValue(sitemapName, pageId));
         if (listener == null) {
             // there is no listener for this page yet, so let's try to create one
-            EList<Widget> widgets = null;
-            widgets = collectWidgets(sitemapName, pageId);
-            if (widgets != null) {
-                listener = new PageChangeListener(sitemapName, pageId, itemUIRegistry, widgets);
-                pageChangeListeners.put(getValue(sitemapName, pageId), listener);
-            }
+            listener = new PageChangeListener(sitemapName, pageId, itemUIRegistry, collectWidgets(sitemapName, pageId));
+            pageChangeListeners.put(getValue(sitemapName, pageId), listener);
         }
         if (listener != null) {
             listener.addCallback(callback);

--- a/bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java
+++ b/bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java
@@ -15,6 +15,7 @@ import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
 
 import org.apache.commons.lang.StringUtils;
+import org.eclipse.emf.common.util.BasicEList;
 import org.eclipse.emf.common.util.EList;
 import org.eclipse.smarthome.io.rest.sitemap.internal.PageChangeListener;
 import org.eclipse.smarthome.io.rest.sitemap.internal.SitemapEvent;
@@ -215,7 +216,7 @@ public class SitemapSubscriptionService implements ModelRepositoryChangeListener
     }
 
     private EList<Widget> collectWidgets(String sitemapName, String pageId) {
-        EList<Widget> widgets = null;
+        EList<Widget> widgets = new BasicEList<Widget>();
 
         Sitemap sitemap = getSitemap(sitemapName);
         if (sitemap != null) {

